From aee992fcc92f9be3db0b47b131f4d601f3f39571 Mon Sep 17 00:00:00 2001
From: Timm Allman <tallman@umass.edu>
Date: Mon, 19 Jan 2015 15:51:05 -0500
Subject: [PATCH 07/14] update to pull from my repo

---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index 0d2bb92..ae296f3 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "src/Heap-Layers"]
 	path = src/Heap-Layers
-	url = https://github.com/emeryberger/Heap-Layers.git
+	url = https://github.com/timm-allman/Heap-Layers.git
-- 
1.9.5.github.0

